server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1842,13 +1835,12 @@ record RootBlobUpdateResult(RepositoryData oldRepositoryData, RepositoryData new
                         }));
                     }
 
-                    // Write the SnapshotInfo blob
-                    executor.execute(
-                        ActionRunnable.run(
-                            allMetaListeners.acquire(),
-                            () -> SNAPSHOT_FORMAT.write(snapshotInfo, blobContainer(), snapshotId.getUUID(), compress)
-                        )
-                    );
+                    // Write the SnapshotInfo blob (we're already on a SNAPSHOT thread so no need to fork this)
```

Comment:
Seems like we're implicitly indicating the network / remote blob store part, so might as well just be explicit. Something like this
```suggestion
                    // Write the SnapshotInfo blob over the network (we're already on a SNAPSHOT thread so no need to fork this)
```

Reply:
It might not be over the network tho, it could be in a `FsRepository`. I'll add "to the repository", maybe that'll help?